sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1070,6 +1107,7 @@ public Instant getWatermark() {
 
     @Override
     public CheckpointMark getCheckpointMark() {
+      reportBacklog();
```

Comment:
This is the first source we're adding metrics to.

It seemed excessive to report backlog after every single event read and this method is called only every so often.

I agree this is a bit hacky though.
@dhalperi is there a more natural place to do this? Do we want to add one in the source API (I'm assuming the answer is no)?